background.js

```diff
@@ -32,8 +32,92 @@ function uninstallBlocker() {
 }
 function onBeforeRequest(aDetails) {
   log('onBeforeRequest', aDetails);
-  launch(aDetails.url);
-  return { cancel: true };
+  var list = configs.exclusionielist.trim().split(/\s+/).filter((aItem) => !!aItem);
+  log('exclusion list: ', list);
+  var redirected = true;
+
+  var matched = false;
+  list.forEach((pattern) => {
```

Comment:
I think a single large regexp like `new RegExp('(' + list.map(matchPatternToRegExp).join('|') + ')')` is better than native JS loop.